### PR TITLE
Add Arviz datatypes doc

### PR DESCRIPTION
**Pull request by canyon289**
Will keep this gist updated with the latest version.

For transparency still got some comments from @ColCarroll that I need to resolve

https://gist.github.com/canyon289/3ca84922cae29f706a36dab3b21ac065

### CODE REVIEW

**Comment by ColCarroll** [comment]
Thanks for this!  Once this is integrated into the docs (I can do that), I think we're just about ready for a release. Also, we can see a preview in the "files changed" section if you don't want to update the gist. 

I think I would change the title to `Introduction to xarray for ArviZ`, and rerun the notebook so the cells are sequential. 

There are a few typos (bayesian practioner -> Bayesian practitioner), but this is definitely an improvement, and others could make PRs on it (in particular, we may want to ping the xarray devs to make sure we are not offending them :) ) Given the two changes above, I would say this is ok to merge, and we can iterate after if we want to.

**Comment by canyon289** [comment]
Great. I'll do the three things of doing a last sweep for typos, title change, and contacting the xarray devs!

**Comment by canyon289** [comment]
Made the updates, I sent an email to Fujii asking if he'd be willing to read this over. If you guys know of an appropriate way to contact the xarray devs in general let me know, not sure if they have a mailing list or something like that

File: doc/notebooks/XarrayforArviZ.ipynb [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    "* Prior Predictive Distribution\n",
+    "* Posterior Predictive Distribution\n",
+    "* Trace data for each of the above\n",
+    "* Sample statistics for each inference run\n",
+    "* Whatever else\n",
+    "\n",
+    "## Why not Pandas Dataframes or Numpy Arrays?\n",
+    "Pandas would be a wonderful data structure, but data from probabilistic programming is naturally high dimensional. That is precisely the niche *xarray* seeks to fill. To add to the complexity ArviZ must handle the data generated from multiple Bayesian Modeling libraries, such as pymc3 and pystan.\n",
+    "\n",
+    "Although seemingly more complex at a glance the Arviz devs believe that the usage of *xarray*, *InferenceData*, and *netcdf* will simply the handling, referencing, and serialization of data generated by MCMC runs."
```
Comment (ahartikainen) [inline doc/notebooks/XarrayforArviZ.ipynb]: simply ? simplify

File: doc/notebooks/XarrayforArviZ.ipynb [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## An introduction to each\n",
+    "To help get familiar with each ArviZ includes some toy datasets. To start an `az.InferenceData` sample can be loaded into memory quite easily."
```
Comment (ahartikainen) [inline doc/notebooks/XarrayforArviZ.ipynb]: each, Ar...

File: doc/notebooks/XarrayforArviZ.ipynb [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    "# Get the observed xarray\n",
+    "observed_data = data.observed_data\n",
+    "observed_data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It should be noted that the observed dataset contains only 8 DataVariables and shares no dimensions or coordinates with the posterior. This difference in sizes is the motivating reason behind *InferenceData*. Rather than force multiple different sized arrays into one array, or force users to manage multiple objects corresponding to different datasets, it is easier to hold references to each *xarray* in an *InferenceData* object."
```
Comment (ahartikainen) [inline doc/notebooks/XarrayforArviZ.ipynb]: Isn't one of the dims the school dimension?

**Comment by canyon289** [comment]
Thanks for the comments at @ahartikainen. They're all helpful. I'll get to this on Sunday in ~2 days

**Comment by ColCarroll** [comment]
I think this is fine to merge if you do @canyon289 , and it can continue to improve via PRs.

**Comment by ColCarroll** [comment]
Oh, I meant if xarray had comments, or if there were other changes you planned to make (or if we wanted to make spelling/grammar nitpicks without dragging this PR out longer).

**Pull request merged**